tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -971,3 +971,42 @@ def test_invalid_multivariate_and_group() -> None:
 def test_group_experimental_warning() -> None:
     with pytest.warns(optuna.exceptions.ExperimentalWarning):
         _ = TPESampler(multivariate=True, group=True)
+
+
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+def test_constant_liar_observation_pairs(direction: str) -> None:
```

Comment:
Open for ideas if there's a better way to test. This test uses private APIs and doesn't test the propagation of the `constant_liar` argument from `__init__` to observation pair extraction.

Reply:
This might be out of the scope of this PR, but I have an idea not for the test but for verification of the algorithm itself. How about plotting the density of GMM like the following setting?

- Consider low-dimensional objective function such as the rosenblock function. 
- For `x: float`, the value of the density of GMM is calculated by `TPESampler._gmm_log_pdf(samples=[x], ...)`.

Reply:
Right, we could perhaps do that to check the algorithm. But I'm wondering if that isn't already verifiable using the notebook in the description. Do you think we can write unit tests based on this?

Reply:
No, I don't think we need to go that far with unit tests. Indeed, as you said, we can see that various points are being evaluated at once, as shown in the colab above.

I thought this would be a good opportunity as the request to plot KDE values has been raised by users before, but this seems to be beyond the scope of this PR. (FYI: https://github.com/optuna/optuna/issues/2600#issuecomment-824663938)